src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -959,7 +959,11 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           doIt
         }
 
-
+        def isJavaOrUniversal(sym: Symbol) = {
+          sym.isJavaDefined ||
+          isUniversalMember(sym) ||
```

Comment:
This includes `wait`, `notify`, `notifyAll`, `clone`, `finalize`. [Dotty does](https://github.com/lampepfl/dotty/blob/811dc19f418f26d6b830bb08bfb49aeee3853a49/compiler/src/dotty/tools/dotc/core/SymDenotations.scala#L950-L957) `sym.owner == defn.AnyClass || sym == defn.Object_clone`, I think we should align with that.

Reply:
addressed it here - https://github.com/scala/scala/pull/8296/commits/e18dd0d592c82ed8a7e1b082bafb0c4c59305f0a